### ISSUE TITLE
concmd("+use") isn't working

### ISSUE THREAD

**Issue by SomeUselessGuy** [wiremod/wire#1789]
concmd("+use") isn't working. wire_expression2_concmd = 1;wire_expression2_concmd_whitelist = "+use,-use". that isn't working only after update

**Comment by TomyLobo** [comment]
Console commands are set without an equals sign and you enter them in the developer console, not E2.

**Comment by OriaOriginal** [comment]
> Console variables are set without an equals sign and you enter them in the developer console, not E2.

On the server where I play, the entity:use() is restricted, so before update we used concmd("+ use"). After this update, it stopped working.

**Comment by TomyLobo** [comment]
That has nothing to do with what I said.

**Comment by CaptainPRICE** [comment]
> Console variables are set without an equals sign and you enter them in the developer console, not E2.
> That has nothing to do with what I said.

I think @OriaOriginal is just copying/pasting the values of the actual convars. That's how console spews it out, with an equal sign.
> ] wire_expression2_concmd 
> "wire_expression2_concmd" = "1" ( def. "0" )

So, if I read this right.. Checking this out on the latest [`master`](becd300f355d857e28ba7c07cbbfdaff9aa58a66) branch:

```golo
# if (!convarnum("wire_expression2_concmd")) { error("concmd is disabled") }
concmd("+use;-use")
interval(1000)
```

I have also tried changing `wire_expression2_concmd_whitelist`. Indeed, yes, **it has stopped from working** properly.

I think I know who...

@thegrb93 *cough* #1748 *cough*

**Comment by CaptainPRICE** [comment]
> Indeed, yes, it has stopped from working properly.

I'm sorry Garrett, lets not rush this time. Ok.
tldr; This is a limitation of Source engine. Your #1748 has nothing to do with this "*issue*". Thus, this is not a bug, this is intended behaviour in engine.

---

### More details
Moving my analysis further down, I have gone back in time before #1748 were merged.
Tried to use exactly the same code, and were surprised - it did not work. Which means the issue is not caused by PR #1748.
Going back to C++ of ConCommand, and found that command buffer is actually queued.
So, I had changed E2 code accordingly to use the following delay/timer mechanism (pressing and releasing `use` command every 500 milliseconds):
```golo
@outputs Use

if (clk("+use") | clk("-use"))
{
    local Command = clkName()
    print("Performing " + Command)
    concmd(Command)
}

Use = !Use
timer(Use ? "+use" : "-use", 500)
```

💥 And it worked, it held for 500ms and then followed by release, and repeating it (in never ending loop - forever).
Travelling back to the future... (Oh, Divran just made a new commit to the [master](28358614fdbabdd8c0d90727575f2e75ee65184c)).
Testing exactly the same E2 code, it works just fine.

So, there it is, at least as #1748 is what I were concerned about.

**Conclusion**: You **can not** have `+use;-use` run in the same tick.

**To proof that**: Use console command itself while nearby & aiming at usable entity (such as a Button), and then submit `+use;-use` into console. You will see it won't process it. No use command will be registered. You will also notice it wouldn't work even if you had 2 split calls (i.e. `concmd("+use") concmd("-use")`), because E2 execution is still running in the same server tick, thus `+use` is cancelled out.

Solution: Fix your E2 code, @OriaOriginal. See mine test E2 code above, you'll want to implement something alike.

/engine-limitation
/not-a-bug
/incompetent-subject-code

**Comment by thegrb93** [comment]
Before 'wait' was removed, you would normally put a 'wait' in between them and it would work. You should be able to use a timer to easily get around this though.

**Comment by thegrb93** [comment]
Oh that wasn't the issue though

**Comment by thegrb93** [comment]
White list is delimited with a comma, he wasn't actually trying to +use;-use, so i think you just wasted your time :P
